pkg/core/object/replicate.go

```diff
@@ -59,6 +62,9 @@ func EncodeReplicationMetaInfo(cID cid.ID, oID, firstPart, previousPart oid.ID,
 	if len(locked) > 0 {
 		kvs = append(kvs, oidsKV(lockedKey, locked))
 	}
+	if typ != objectsdk.TypeRegular {
+		kvs = append(kvs, kv(typeKey, uint32(typ)))
```

Comment:
although `typ` is `uint32`, lets to cast to `int32`. It will be `int32` soon (im also unhappy bout this, but wcyd). We'll most likely forget to change conversion later

Reply:
ok